src/dataIntegration/transcriptomics/preprocessing/selectGeneFromGPR.m

```diff
@@ -19,55 +19,55 @@
 %                   for OR
 %
 % OUTPUTS:
-%   expressionCol:  reaction expression, corresponding to model.rxns.
+%   expressionRxns:  reaction expression, corresponding to model.rxns.
 %                   No gene-expression data and orphan reactions will
 %                   be given a value of -1.
+%          .rxnSig - significance for each rxn associated with each
+%                     reaction
+%           .rxnExp - expression for each rxn associated with each
+%                     reaction
 %
-% AUTHOR: Anne Richelle, May 2017
-
+% ORIGINAL AUTHOR: Anne Richelle, May 2017
+%       - Modified by Chaitra Sarathy, Aug 2019, to include significance
+%       value as additional input
 
 if ~exist('minSum','var')
     minSum = false;
 end
-gene_used={};
-for i=1:length(model.rxns)
-	gene_used{i}='';
-end
-    
 % -1 means unknown/no data
-expressionCol = -1*ones(length(model.rxns),1); 
+expressionRxns.rxnSig = zeros(length(model.rxns),1); 
+expressionRxns.rxnExp = -1*ones(length(model.rxns),1); 
 for i = 1:length(model.rxns)
```

Comment:
Please modify the code so that it allows using significance as an optional input.

Reply:
Thank you @tpfau for your comments. It is a good point to keep the new variables optional and not break existing code. I will modify the function to take significance value as optional